.github/workflows/ci.yml

```
@@ -0,0 +1,15 @@
+name: CI
+on: [push]
+jobs:
+  format-check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - run: npm ci
+      - run: npm run format-check
+  test-compile:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - run: npm ci
+      - run: npm run test-compile
```

This is still duplicating the same initialisation twice on two runners, let's remove that wasted work to make it faster and to consume less compute
```suggestion
  test:
    runs-on: ubuntu-latest
    steps:
      - run: npm ci
      - run: npm run test-compile
      - run: npm run format-check
```